src/components/selection-list/selection-list.tsx

```diff
@@ -0,0 +1,136 @@
+    let current: HTMLElement | null = startAt;
+    while (current && !predicate(current)) {
+        current = current.parentElement;
+    }
+    return current;
+}
+
+export interface ItemRendererProps {
+    focused: boolean;
+    selected: boolean;
+    item: any;
+};
+
+export interface DefaultItemRendererProps extends ItemRendererProps {
+    item: {
+        value?: string;
+        label?: string;
+        hidden?: boolean;
+        disabled?: boolean;
+    }
+}
+
+const DefaultItemRenderer: React.SFC<DefaultItemRendererProps> = SBStateless(props => {
+    if (props.item === divider) {
+        return <div className="divider" data-automation-id="DIVIDER"></div>;
+    }
+
+    if (props.item.hidden) {
+        return null;
+    }
+
+    return (
+        <div
+            className="list-item"
+            data-value={props.item.disabled! ? null : props.item.value}
+            cssStates={{
+                focused: props.focused,
+                selected: props.selected,
+                disabled: props.item.disabled!
+            }}>
+            {props.item.label}
+        </div>
+    );
+}, style);
+
+export type SelectionItem = string | object;
+
+export interface OptionList {
+    dataSource?: SelectionItem[],
+    dataSchema?: {},
+    itemRenderer?: React.ComponentType<ItemRendererProps>
+}
+
+export interface SelectionListProps extends OptionList {
+    value?: string;
+    onChange?: (value: string) => void;
+    style?: any;
+    children?: any
+}
+
+@SBComponent(style)
+export class SelectionList extends React.Component<SelectionListProps, {}> {
+    static defaultProps: SelectionListProps = {
+        dataSource: [],
+        itemRenderer: DefaultItemRenderer,
+        onChange: () => {}
+    };
+
+    normalizeItem(item: SelectionItem): {[index: string]: any} {
+        if (item === divider) {
+            return divider;
+        }
+
+        if (typeof item === 'string') {
+            item = {value: item, label: item};
+        }
+
+        return this.props.dataSchema ?
+            renameKeys(item, this.props.dataSchema) : item;
+    }
+
+    renderItem(item: SelectionItem, index: number) {
+        const ItemRenderer = this.props.itemRenderer!;
+        const normalized = this.normalizeItem(item);
+        return <ItemRenderer
+            key={index}
+            focused={false}
+            selected={normalized.value === this.props.value}
+            item={normalized} />;
+    }
+
+    handleClick: React.EventHandler<React.MouseEvent<HTMLElement>> = (event) => {
+        const root = ReactDOM.findDOMNode(this);
+        const item = closestElementMatching(
+            (el) => el.parentElement === root,
+            event.target as HTMLElement
+        );
+        if (!item) {
+            return;
+        }
+        const value = item.dataset.value;
+        if (value === undefined || value === this.props.value) {
+            return;
+        }
+        this.props.onChange!(value);
+    };
```

Comment:
@tyv 
> I don't want to, why? I want to add handler to each item

I'm not sure what you mean by that. If you don't agree with the spec, it's no use discussing implementation, go and comment on the spec. And if you don't like the implementation, you need to be more elaborate than simply saying "attributes are bad", "use onClick". What exactly is the problem with attributes? How exactly do you propose to solve it using onClick? Does that require an API change?

@VKobeliatsky 
> DOM operations are a bottle neck

Depends on the DOM operation. Custom attributes are cheap, and probably cause less garbage collection than closures. Although I don't think there's a noticeable performance difference between the two, whether it's an initial render or not.